packages/lodestar/src/sync/sync.ts

```diff
@@ -199,15 +199,27 @@ export class BeaconSync implements IBeaconSync {
   private onUnknownBlockRoot = async (root: Root): Promise<void> => {
     const peerBalancer = new RoundRobinArray(this.getPeers());
     // eslint-disable-next-line no-constant-condition
-    while (true) {
+    let retry = 0;
+    const maxRetry = this.getPeers().length;
+    while (retry < maxRetry) {
```

Comment:
I think previous solution should be fine as long as RoundRobinArray returns null once we iterate over all peers.

If you wanna improve, we can maybe try single peer multiple times but this could be awkward if we start waiting on response timeouts of peer

Reply:
> I think previous solution should be fine as long as RoundRobinArray returns null once we iterate over all peers.

yeah but currently `RoundRobinArray` does not return null in that case. I don't really want to change `RoundRobinArray` since our sync is stable enough, wdyt @mpetrunic ?

Reply:
yup, makes sense